src/iadmin.cpp

```diff
@@ -1620,8 +1620,22 @@ doCommand( char *cmdToken[], rodsArguments_t* _rodsArgs = 0 ) {
     }
 
     if ( strcmp( cmdToken[0], "mkgroup" ) == 0 ) {
-        generalAdmin( 0, "add", "user", cmdToken[1], "rodsgroup",
-                      "", "", "", "", "", "" );
+        const auto version = irods::to_version(Conn->svrVersion->relVersion);
+        const bool using_add_group = (version && version.value() > irods::version{4, 3, 3});
+        const char* entity_type = using_add_group ? "group" : "user";
+        // If we are adding a group, we don't need to specify a user type.
+        const char* user_type = using_add_group ? "" : "rodsgroup";
+        generalAdmin(0,
+                     "add",
+                     const_cast<char*>(entity_type),
+                     cmdToken[1],
+                     const_cast<char*>(user_type),
```

Comment:
Could use `std::string::data` to remove the need for `const_cast`.

Reply:
See https://github.com/irods/irods_client_icommands/pull/532#discussion_r1951291560